src/main/webapp/js/build-message-div.js

```diff
@@ -42,7 +42,71 @@ function formatTimestamp(messageTime){
           timestamp.getHours()+":"+timestamp.getMinutes()+":"+timestamp.getSeconds();
 }
 
-function buildMessageInTimeline(message, flip){
+function createDeleteButton(deleteConfirmationDiv){
+  var btn = document.createElement('button');
+  btn.type='button';
+  btn.classList.add('delete');
+  var icon = document.createElement('i');
+  icon.classList.add('fa','fa-trash');
+  btn.appendChild(icon);
+  btn.onclick = function(){
+    deleteConfirmationDiv.classList.remove('hide');
+    deleteConfirmationDiv.classList.add('show');
+  };
+  return btn;
+}
+
+function deletePost(messageBlock) {
+  messageBlock.style.minHeight = '0';
+  messageBlock.style.margin = '0';
+  messageBlock.style.height = messageBlock.scrollHeight + 'px';
+  window.setTimeout(function(){messageBlock.style.height='0';}, 1);
+  window.setTimeout(function(){
+    var parent = messageBlock.parentNode;
+    var next = messageBlock.nextSibling;
+    parent.removeChild(messageBlock);
+    if(!parent.hasChildNodes()) {
+      document.getElementById('noPosts').hidden = false;
+    } else {
+      while(next != null) {
+        if(next.classList.contains('timeline-inverted')) {
+          next.classList.remove('timeline-inverted');
+        } else {
+          next.classList.add('timeline-inverted');
+        }
+        next = next.nextSibling;
+      }
+    }
+  }, 200);
+}
+
+function createDeleteConfirmationDiv(){
+  var div = document.createElement('div');
+  div.classList.add('deleteConfirmation','alert','alert-warning','hide');
+  var closeBtn = document.createElement('button');
+  closeBtn.type = 'button';
+  closeBtn.classList.add('close');
+  closeBtn.innerHTML = '&times;';
+  closeBtn.onclick = function(){
+    div.classList.remove('show');
+    div.classList.add('hide');
+  };
+  var text = document.createTextNode('Are you sure you want to delete this post?');
```

Comment:
```suggestion
  var text = document.createTextNode('Are you sure you want to delete this post? ');
```
Just so that the `Delete` link doesn't appear right next to the question mark